SC_redesign/Crop_and_Soil/manager/field_manager.py

```
@@ -239,3 +250,115 @@ def _setup_crop_schedules(crop_config: Dict) -> List[CropSchedule]:
+        config_dictionary["soil_layers"] = soil_layers
+
+        soil_data = SoilData(field_size=field_size, **config_dictionary)
+        return Soil(soil_data=soil_data)
+
+    @staticmethod
+    def _setup_soil_layer(field_size: float, top_depth: float, sand: float, silt: float, initial_residue: float,
+                          fresh_nitrogen_mineralization_rate: float, layer_config: Dict) -> LayerData:
+        """
+        Initializes a LayerData instance to be added to a SoilData object.
+
+        Parameters
+        ----------
+        field_size : float
+            Size of the field that contains the soil layer being created (ha)
+        top_depth : float
+            Depth of top of the soil layer beneath the surface (mm)
+        sand : float
+            Sand content expressed as percent of soil in this layer (unitless)
+        silt : float
+            Silt content expressed as percent of soil in this layer (unitless)
+        initial_residue : float
+            Amount of residue on the soil surface when this soil layer is initialized (kg / ha)
+        fresh_nitrogen_mineralization_rate : float
+            Rate at which fresh organic nitrogen mineralizes (unitless)
+        layer_config : Dict
+            Contains all the specifications for a layer of soil.
+
+        Returns
+        -------
+        LayerData
+            LayerData instance configured with provided data.
+
+        Notes
+        -----
+        Whoever wrote the JSON's for soil profile inputs wrote "N03" (the digit zero) instead of "NO3" (the letter 'O'),
+        and that is why it is written with a zero and not the letter here.
+
+        """
+        config_dictionary = {}
+
+        config_dictionary["bottom_depth"] = layer_config["bottom_depth"]
+        config_dictionary["wilting_point_water_concentration"] = layer_config.get("wilting_point")
+        config_dictionary["field_capacity_water_concentration"] = layer_config.get("field_capacity")
```

It looks like when you use square brackets around a field, you assume that that field should be there. When you use `get()`, it means optional and can return None. I think you should raise an error every time a required field is absent.